gensim/models/_fasttext_bin.py

```diff
@@ -366,5 +386,266 @@ def _backslashreplace_backport(ex):
+    [FastText::signModel](https://github.com/facebookresearch/fastText/blob/master/src/fasttext.cc)
+
+    Parameters
+    ----------
+    fout: writeable binary stream
+    """
+    fout.write(_FASTTEXT_FILEFORMAT_MAGIC.tobytes())
+    fout.write(_FASTTEXT_VERSION.tobytes())
+
+
+def _conv_field_to_bytes(field_value, field_type):
+    """
+    Auxiliary function that converts `field_value` to bytes based on request `field_type`,
+    for saving to the binary file.
+
+    Parameters
+    ----------
+    field_value: numerical
+        contains arguments of the string and start/end indexes of the bad portion.
+
+    field_type: str
+        currently supported `field_types` are `i` for 32-bit integer and `d` for 64-bit float
+    """
+    if field_type == 'i':
+        return (np.int32(field_value).tobytes())
+    elif field_type == 'd':
+        return (np.float64(field_value).tobytes())
+    else:
+        raise NotImplementedError('Currently conversion to "%s" type is not implemmented.' % field_type)
+
+
+def _get_field_from_model(model, field):
+    """
+    Extract `field` from `model`.
+
+    Parameters
+    ----------
+    model: gensim.models.fasttext.FastText
+        model from which `field` is extracted
+    field: str
+        requested field name, fields are listed in the `_NEW_HEADER_FORMAT` list
+    """
+    if field == 'bucket':
+        res = model.trainables.bucket
+    elif field == 'dim':
+        res = model.vector_size
+    elif field == 'epoch':
+        res = model.epochs
+    elif field == 'loss':
+        # `loss` => hs: 1, ns: 2, softmax: 3, ova-vs-all: 4
+        # ns = negative sampling loss (default)
+        # hs = hierarchical softmax loss
+        # softmax =  softmax loss
+        # one-vs-all = one vs all loss (supervised)
+        if model.hs == 1:
+            res = 1
+        elif model.hs == 0:
+            res = 2
+        elif model.hs == 0 and model.negative == 0:
+            res = 1
+    elif field == 'maxn':
+        res = model.wv.max_n
+    elif field == 'minn':
+        res = model.wv.min_n
+    elif field == 'min_count':
+        res = model.vocabulary.min_count
+    elif field == 'model':
+        # `model` => cbow:1, sg:2, sup:3
+        # cbow = continous bag of words (default)
+        # sg = skip-gram
+        # sup = supervised
+        res = 2 if model.sg == 1 else 1
+    elif field == 'neg':
+        res = model.negative
+    elif field == 't':
+        res = model.vocabulary.sample
+    elif field == 'word_ngrams':
+        # This is skipped in gensim loading settig, using the default from FB C++ code
+        res = 1
+    elif field == 'ws':
+        res = model.window
+    elif field == 'lr_update_rate':
+        # This is skipped in gensim loading settig, using the default from FB C++ code
+        res = 100
+    else:
+        msg = 'Extraction of header field "' + field + '" from Gensim FastText object not implemmented.'
+        raise NotImplementedError(msg)
+    return res
```

Comment:
Is there a reason why we're not utilizing early returns here?

Reply:
Corrected.